### PR TITLE
Add revoke authorisation implementation for Gitlab

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
Add revoke authorisation implementation for Gitlab.

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/22832

### How to test this PR?

1. Deploy Che from the PR image: `quay.io/eclipse/che-server:pr-667`.
2. Apply dashboard image from the related PR: `quay.io/eclipse/che-dashboard:pr-1068`.
3. Setup [gitlab oauth](https://eclipse.dev/che/docs/stable/administration-guide/configuring-oauth-2-for-gitlab/) for onprem or SAAS version.
4. Start a workspace from the gitlab repository with a devfile, accept the authentication request.
5. Go to dashboard -> User Preferences -> Git Services tab and revoke the gitlab authorisation.
6. Delete the workspace.
7. Start the workspace again.

See: The authorisation request appears as it was rejected by the revoke action.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by tolusha** [comment]
Could you share api documentation?

File: wsmaster/che-core-api-auth-gitlab/src/main/java/org/eclipse/che/security/oauth/GitLabOAuthAuthenticator.java [hunk 1]
```diff
@@ -97,6 +102,28 @@ public OAuthToken getToken(String userId) throws IOException {
...
+                    format(
+                        "%s/oauth/revoke?client_id=%s&client_secret=%s&token=%s",
+                        gitlabEndpoint, clientId, clientSecret, token)))
+            .POST(HttpRequest.BodyPublishers.noBody())
+            .build();
+
+    try {
+      HttpResponse<InputStream> response =
+          client.send(request, HttpResponse.BodyHandlers.ofInputStream());
+      return response.statusCode() == 200;
```
Comment (tolusha) [inline wsmaster/che-core-api-auth-gitlab/src/main/java/org/eclipse/che/security/oauth/GitLabOAuthAuthenticator.java:121]: What does it return if token not found (already removed) ?

Comment (vinokurig) [reply wsmaster/che-core-api-auth-gitlab/src/main/java/org/eclipse/che/security/oauth/GitLabOAuthAuthenticator.java:121]: We can not remove an oauth token from Gitlab. If we manually revoke the authorisation from Gitlab, and then execute the revoke action from che dashboard, we will have status code 200.

**Review by tolusha — APPROVED** [review]

**Review by dmytro-ndp — APPROVED** [review]
Tested successfully following test scenario from PR description using GitLab Server.
Screencast: 
[screencast-bpconcjcammlapcogcnnelfmaeghhagj-2024.03.16-02_00_29.webm](https://github.com/eclipse-che/che-server/assets/1197777/d9be1d68-78eb-4d7b-b252-389d348c95bf)

Additional verification has passed as well: no authorization request appeared on new workspace start if not revoke gitlab authorization.
Screencast: 
[screencast-bpconcjcammlapcogcnnelfmaeghhagj-2024.03.15-17_27_56.webm](https://github.com/eclipse-che/che-server/assets/1197777/8f879072-d8ac-4d63-9a99-bcafaed8d121)

**Review by dmytro-ndp — CHANGES_REQUESTED** [review]
I have noticed later, that if revoke authorization and refresh "Git Services" page, it shows "Authorized" status again:  
[screencast-bpconcjcammlapcogcnnelfmaeghhagj-2024.03.16-02_09_36.webm](https://github.com/eclipse-che/che-server/assets/1197777/4a1b7d0b-d8ae-40bf-b8ee-79fa75eb46d2)

@vinokurig : could you please take a look?

**Comment by vinokurig** [comment]
@dmytro-ndp That's a known issue: https://github.com/eclipse/che/issues/22790

**Comment by dmytro-ndp** [comment]
@vinokurig : I see.
Thanks for the explanation.

**Review by dmytro-ndp — APPROVED** [review]

**Pull request merged**